README.md

```diff
@@ -0,0 +1,138 @@
+# Remotely Signed S3 Requests
+This is a library and tool designed to support doing Amazon S3 uploads and
+downloads with request signing occuring on a different machine than the machine
+performing the actual upload or download.
+
+With this library, you can allow untrusted hosts to upload to an object which
+you've protected with whichever authentication and authorization scheme you
+choose.
+
+There is support for both multipart and singlepart uploads and downloads.  The
+high level interfaces provided will automatically select which version to use.
+
+## Architecture
+This project is divided into a server-side component called the `Controller`.
+Instances of the `Controller` know how to genereate and sign all the methods
+involved in uploading files to S3.  `Controller`s also know how to run methods
+which must be run from the server.  The methods which must be run from the
+server are the initate, complete and abort methods of the multipart uploads.
+
+The `Controller` class has the following methods:
+
+* `new Controller(region, runner)`: Create a `Controller` instance for the
+  region specified by `region`.  The `runner` parameter is a function with the
+  signature `(request, body)` which must take a standard interchange format
+  object and returns an object like `Runner.prototype.run`
```

Comment:
I suspect that I need to update this comment, i changed how the Runner is passed in.